src/effects/effectsmanager.cpp

```diff
@@ -133,6 +133,55 @@ void EffectsManager::registerOutputChannel(const ChannelHandleAndGroup& handle_g
     m_registeredOutputChannels.insert(handle_group);
 }
 
+void EffectsManager::loadStandardEffect(const int iChainSlotNumber,
+        const int iEffectSlotNumber, const QString& effectId,
+        EffectBackendType backendType) {
+    auto pChainSlot = getStandardEffectChainSlot(iChainSlotNumber);
+    if (pChainSlot) {
+        loadEffect(pChainSlot, iEffectSlotNumber, effectId, backendType);
+    }
+}
+
+void EffectsManager::loadOutputEffect(const int iEffectSlotNumber,
+    const QString& effectId, EffectBackendType backendType) {
+    if (m_outputEffectChainSlot) {
+        loadEffect(m_outputEffectChainSlot, iEffectSlotNumber, effectId, backendType);
+    }
+}
+
+void EffectsManager::loadQuickEffect(const QString& group,
+        const int iEffectSlotNumber, const QString& effectId,
+        EffectBackendType backendType) {
+    auto pChainSlot = getQuickEffectChainSlot(group);
+    if (pChainSlot) {
+        loadEffect(pChainSlot, iEffectSlotNumber, effectId, backendType);
+    }
+}
+
+void EffectsManager::loadEqualizerEffect(const QString& group,
+        const int iEffectSlotNumber, const QString& effectId,
+        EffectBackendType backendType) {
+    auto pChainSlot = getEqualizerEffectChainSlot(group);
+    if (pChainSlot) {
+        loadEffect(pChainSlot, iEffectSlotNumber, effectId, backendType);
+    }
+}
+
+void EffectsManager::loadEffect(EffectChainSlotPointer pChainSlot,
+        const int iEffectSlotNumber, const QString& effectId,
+        EffectBackendType backendType) {
+    foreach (EffectsBackend* pBackend, m_effectsBackends) {
```

Comment:
Use standard C++ `for (EffectsBackend* pBackend : m_effectsBackends)` instead of Qt `foreach` macro... although this loop wouldn't be necessary if EffectsManager stored a QHash of EffectsBackends indexed by EffectBackendType.